client/my-sites/site-settings/custom-post-types-fieldset/index.jsx

```diff
@@ -0,0 +1,143 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isRequestingPostTypes, getPostTypes } from 'state/post-types/selectors';
+import localize from 'lib/mixins/i18n/localize';
+import QueryPostTypes from 'components/data/query-post-types';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+
+class CustomPostTypesFieldset extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.boundOnChange = this.onChange.bind( this );
+		this.state = { hadOnceEnabled: {} };
+	}
+
+	componentWillMount() {
+		this.updateHadOnceEnabled();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.setState( { hadOnceEnabled: {} }, this.updateHadOnceEnabled );
+		} else {
+			this.updateHadOnceEnabled( nextProps );
+		}
+	}
+
+	updateHadOnceEnabled( props = this.props ) {
+		this.setState( {
+			hadOnceEnabled: [
+				'jetpack-testimonial',
+				'jetpack-portfolio'
+			].reduce( ( memo, postType ) => {
+				const valueKey = this.getPostTypeValueKey( postType );
+				memo[ postType ] = this.state.hadOnceEnabled[ postType ] || props.value[ valueKey ];
+				return memo;
+			}, {} )
+		} );
+	}
+
+	hasDefaultPostTypeEnabled( postType ) {
+		return (
+			! this.props.requestingSettings &&
+			this.props.postTypes &&
+			! this.state.hadOnceEnabled[ postType ] &&
+			! this.props.value[ this.getPostTypeValueKey( postType ) ] &&
+			!! this.props.postTypes[ postType ]
+		);
+	}
+
+	getPostTypeValueKey( postType ) {
+		if ( 'jetpack-testimonial' === postType ) {
+			return 'jetpack_testimonial';
+		}
+
+		if ( 'jetpack-portfolio' === postType ) {
+			return 'jetpack_portfolio';
+		}
+	}
+
+	onChange( event ) {
+		this.props.onChange( {
+			[ event.target.name ]: event.target.checked
+		} );
+	}
+
+	render() {
+		const {
+			translate, siteId, requestingPostTypes, requestingSettings,
+			value, recordEvent, className
+		} = this.props;
+
+		return (
+			<FormFieldset className={ className }>
+				{ siteId && (
+					<QueryPostTypes siteId={ siteId } />
+				) }
+				<FormLabel>{ translate( 'Custom Post Types' ) }</FormLabel>
```

Comment:
It looks like the text here is form wp-admin, where the section is titled "Custom Content Types" - as does the linked documentation http://jetpack.com/support/custom-content-types/.  Should we retain that nomenclature here?